zenoh/src/api/encoding.rs

```diff
@@ -836,24 +836,17 @@ impl EncodingMapping for serde_pickle::Value {
     const ENCODING: Encoding = Encoding::APPLICATION_PYTHON_SERIALIZED_OBJECT;
 }
 
-pub trait EncodingInternals {
-    fn id(&self) -> u16;
-
-    fn schema(&self) -> Option<&ZSlice>;
-
-    fn new(id: u16, schema: Option<ZSlice>) -> Self;
-}
-
-impl EncodingInternals for Encoding {
-    fn id(&self) -> u16 {
+impl Encoding {
```

Comment:
So, if I understand, there is no way for the user to retrieve the id or the schema of its encoding. Is there some interest to expose it at all if use cannot work with it?

Reply:
I mean, if the Rust user cannot have any info about the encoding present in the sample he receives, why exposing it?

Reply:
This was exposed by request from @DariusIMP for kotlin binding. We don't provide schema ids as part of user API, but bindings may need this internal detail.
User can get/set encoding as string or as predefined constant. Internal representation of this encoding string is not exposed

Reply:
The encoding is created/converted to/from `String`. That's the interface exposed to the user. As it stands now, that interface is implemented. Am I missing something?

Reply:
No, I was missing it, sorry. You can close the thread. By the way the discussion about https://github.com/eclipse-zenoh/zenoh/issues/1078#issuecomment-2145224412 is still open.